File: src/View/RunExample.java
package View;

import Controller.Controller;
import Exception.MyStatementExecutionException;
/**
 * Created by vladc on 08.11.2016.
 */
public class RunExample extends Command {
    private Controller ctrl;

    public RunExample(String key, String description, Controller ctrl) {
        super(key, description);
        this.ctrl = ctrl;
    }

    @Override
    public void execute() {
        try {
            ctrl.allStep();
        }
        catch (MyStatementExecutionException e) {
            // todo: treat the exceptions that may occur
            e.printStackTrace();
        }
    }
}

File: Toy-Language-Interpreter_C#/Toy-Language-Interpreter/View/RunExample.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Toy_Language_Interpreter.Controller;

namespace Toy_Language_Interpreter.View
{
    class RunExample : Command
    {
        private MyController ctrl;

        public RunExample(string key, string description, MyController ctrl) : base(key, description)
        {
            this.ctrl = ctrl;
        }

        public override void Execute()
        {
            try
            {
                ctrl.AllStep();
            }
            catch (Exception e) // MyStatementExecutionException
            {
                Console.WriteLine(e.ToString());
                // todo: treat this exception
            }
        }
    }
}
